clients/drcachesim/tools/trace_filter.cpp

```diff
@@ -0,0 +1,217 @@
+/* **********************************************************
+ * Copyright (c) 2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include <fstream>
+#include <thread>
+#include "directory_iterator.h"
+#include "reader/trace_entry_file_reader.h"
+#include "trace_entry.h"
+#ifdef HAS_ZLIB
+#    include "reader/compressed_file_reader.h"
+#    include "common/gzip_ostream.h"
+#endif
+#include "trace_filter.h"
+
+typedef trace_entry_file_reader_t<std::ifstream> default_trace_entry_file_reader_t;
+
+trace_filter_t::trace_filter_t(const std::string &trace_dir,
+                               const std::string &output_dir, int worker_count,
+                               int verbosity)
+    : verbosity_(verbosity)
+    , worker_count_(worker_count)
+    , trace_dir_(trace_dir)
+    , output_dir_(output_dir)
+{
+    if (!init_file_reader_writer(trace_dir, output_dir, verbosity))
+        error_string_ = "Could not initialize trace reader and writer";
+    UNUSED(verbosity_);
+    UNUSED(output_prefix_);
```

Comment:
?  This is not part of this function or class?

Reply:
In some builds, Clang complained about these variables being unused in non-debug mode.